### ISSUE TITLE
kernel.value not using updated hyperparameters with HODLR solver

### ISSUE THREAD

**Issue by ToddSmall** [dfm/george#30]
I'm hitting what I think may be bug in which the kernel.value method is somehow not using updated kernel parameters when using the HODLR solver.

Here's what I'm doing. I train a model using the HODLR solver. I check that the kernel hyperparameters have been updated from their initial values. I then call kernel.value and find that the result is using the kernel with the initial values.

I've made a [sample script](https://gist.github.com/ToddSmall/3d9e1099a7d5b5db6ddc) that reproduces the bug. When you run the script, it prints out the log-likelihood of the trained model, the kernel with trained hyperparameters, and the main diagonal of the covariance matrix. If you run the script using the basic solver

```
python george-bug.py --solver basic
```

the values of the main diagonal are all 1.257, which is, correctly, the sum of the _trained_ ConstantKernel variance and the WhiteKernel.  However, if you run the script using the HODLR solver

```
python george-bug.py --solver HODLR
```

the values of the main diagonal are all 2.0, which is, incorrectly, the sum of the _initial_ ConstantKernel variance and the WhiteKernel variance.

For the record, I'm using Python 2.7 and George 0.2.1 on an old version (Lucid) of Ubuntu.

**Comment by dfm** [comment]
Sorry that I never responded to this. For posterity:

I think that the issue here was actually just that the gradients computed using the approximate HODLR solver are noisy and the optimization doesn't converge. I'm still finding this to be true with the current version of george, but I don't think that it's a bug - just an annoyance of approximate methods. For this example, I found that numerical derivatives (not providing a `jac` argument to minimize) were more stable so perhaps there's something that could be done to make this more reliable, but I'm going to close this for now.
